common-methods.gradle

```diff
@@ -0,0 +1,131 @@
+    LEANPLUM_SDK_VERSION = "$System.env.ANDROID_VERSION_STRING"
+
+    PROGUARD_FILES = 'proguard-rules.pro'
+    CONSUMER_PROGUARD_FILES = 'consumer-proguard-rules.pro'
+
+    pomWithXml = this.&pomWithXml
+    otherNameForMethod2 = this.&commonMethod2
+    publishing_task = this.&publishing_task
+    pomConfig = this.&pomConfig
+}
+
+def publishing_task(libraryArtifactId, packageName) {
+
+    task makeJar(type: Copy) {
+        from('build/intermediates/bundles/release/')
+        into('build/outputs/jar/')
+        include('classes.jar')
+        rename('classes.jar', packageName + '-release.jar')
+    }
+
+    publishing {
+        publications {
+            aar(MavenPublication) {
+                groupId LEANPLUM_GROUP_ID
+                version LEANPLUM_SDK_VERSION
+                artifactId libraryArtifactId
+                artifact("$buildDir/outputs/aar/" + packageName + "-release.aar")
+
+                // The publication doesn't know about our dependencies,
+                // so we have to manually add them to the pom.
+                pomWithXml(pom)
+            }
+            jar(MavenPublication) {
+                groupId LEANPLUM_GROUP_ID
+                version LEANPLUM_SDK_VERSION
+                artifactId libraryArtifactId
+
+                artifact("$buildDir/outputs/jar/" + packageName + "-release.jar")
+
+                // The publication doesn't know about our dependencies,
+                // so we have to manually add them to the pom.
+                pomWithXml(pom)
+            }
+        }
+    }
+
+    if (project.hasProperty('artifactoryUsername') && project.hasProperty('artifactoryPassword')) {
+        artifactory {
+            contextUrl = 'http://artifactory-upload.leanplum.com'
+            publish {
+                repository {
+                    repoKey = 'libs-snapshot-local'
+
+                    username = artifactoryUsername
+                    password = artifactoryPassword
+                }
+                defaults {
+                    publications('aar', 'jar')
+                    publishArtifacts = true
+
+                    properties = ['qa.level': 'basic', 'q.os': 'android', 'dev.team': 'core']
+                    publishPom = true
+                }
+            }
+        }
+    }
+}
+
+def pomWithXml(pom) {
+    pom.withXml {
+        def dependenciesNode = asNode().appendNode('dependencies')
+        // Iterate over the compile dependencies (we don't want the test ones),
+        // adding a <dependency> node for each.
+        configurations.api.allDependencies.each {
+            pomConfig(it, dependenciesNode)
+        }
+        configurations.releaseApi.allDependencies.each {
+            pomConfig(it, dependenciesNode)
+        }
+    }
+}
+
+def pomConfig(itParam, dependenciesNode) {
+    if (itParam.group != null && itParam.name != null) {
+        def dependencyNode = dependenciesNode.appendNode('dependency')
+        if (itParam.name == 'AndroidSDKCore') {
```

Comment:
Since these are all the same except for 'artifactId', could define the different arftifactId values in each individual gradle and pass in as argument.

Reply:
'AndroidSDKCore' this is a name of dependency of a project.  There are only 5 possible dependencies.